scipy/interpolate/tests/test_bsplines.py

```diff
@@ -472,6 +471,34 @@ def run_design_matrix_tests(n, k, bc_type):
         for k in range(2, 7):
             run_design_matrix_tests(n, k, "periodic")
 
+    @pytest.mark.parametrize('extrapolate', [False, True, 'periodic'])
+    @pytest.mark.parametrize('order', range(5))
+    def test_design_matrix_same_as_BSpline_call(self, extrapolate, order):
+        """Test that design_matrix(x) is equivalent to BSpline(..)(x)."""
+        np.random.seed(1234)
+        x = np.random.random_sample(10 * (order + 1))
+        xmin, xmax = np.amin(x), np.amax(x)
+        k = order
```

Comment:
Minor nit: I belive this is degree not order (as in a cubic has degree 3 and the order 4). Somebody went through the codebase and changed a bunch of orders for degrees a while ago. Maybe best to at least agree on the notation.